repo/open.go

```diff
@@ -193,7 +205,14 @@ func writeCacheMarker(cacheDir string) error {
 	}
 
 	markerFile := filepath.Join(cacheDir, CacheDirMarkerFile)
-	if _, err := os.Stat(markerFile); !os.IsNotExist(err) {
+
+	st, err := os.Stat(markerFile)
+	if err == nil && st.Size() >= int64(len(cacheDirMarkerContents)) {
```

Comment:
```suggestion
	if err == nil && st.Size() >= int64(len(CacheDirMarkerHeader)) {
```
It may be a good idea to check the signature header here.
Otherwise this is assuming that the header is correct and it will be skipped in `ignorefs`.
While unlikely, it may be that the contents don't match.

The implemented behavior is to overwrite the file if it is shorter or inconsistent. That's OK, just noting it.